src/routers/handlers/choose_your_package_accounts/choose.your.package.accounts.ts

```diff
@@ -1,61 +1,85 @@
 }
 
+const templatePath = "router_views/choose_your_package_accounts/choose_your_package_accounts";
+
 export class ChooseYourPackageAccountsHandler extends GenericHandler {
-    private packageAccountsItems = getPackageItems();
     constructor() {
         super({
             title: "What package accounts are you submitting?",
             backURL: null
         });
     }
 
-    async executePost (
-        req: Request,
-        _response: Response
-    ): Promise<ChooseYourPackageAccountsViewData> {
-        logger.info(`post request made for ${this.constructor.name}`);
+    getViewData(req: Request): ChooseYourPackageAccountsViewData {
         super.populateViewData(req);
         const companyNumber = getCompanyNumber(req.session);
-        const packageTypeChoice = req.body;
-
-        if ("value" in packageTypeChoice && packageTypeChoice["value"] in PackageTypeDetails){
-            setPackageType(req.session, packageTypeChoice.value);
-        } else {
-            this.baseViewData.errors.packageAccountsError = new Error("Package Accounts type must be set");
-        }
-
-        this.baseViewData.backURL = encodeURIComponent(`${PrefixedUrls.CONFIRM_COMPANY}?companyNumber=${companyNumber}`);
 
         return {
             ...this.baseViewData,
-            packageAccountsItems: this.packageAccountsItems
+            packageTypeFieldName,
+            backURL: `${PrefixedUrls.CONFIRM_COMPANY}?companyNumber=${companyNumber}`,
+            packageAccountsItems: getPackageTypeOptionsRadioButtonData()
         };
     }
 
     async executeGet(
         req: Request,
         _response: Response
-    ): Promise<ChooseYourPackageAccountsViewData>{
-        logger.info(`get request made for ${this.constructor.name}`);
-        super.populateViewData(req);
-        const companyNumber = getCompanyNumber(req.session);
+    ): Promise<ViewModel<ChooseYourPackageAccountsViewData>>{
+        logger.info(`Rendering package accounts type chooser page`);
 
-        this.baseViewData.backURL = encodeURIComponent(`${PrefixedUrls.COMPANY_SEARCH}?companyNumber=${companyNumber}`);
+        return {
+            templatePath,
+            viewData: this.getViewData(req)
+        };
+    }
+
+
+    async executePost (
+        req: Request,
+        _response: Response
+    ): Promise<ViewModel<ChooseYourPackageAccountsViewData> | Redirect> {
+        logger.info(`Handling post request for package accounts chooser page`);
+
+        const viewData = this.getViewData(req);
+
+        const packageTypeChoice = req.body[packageTypeFieldName];
+        logger.debug(`Submitted choice: ${JSON.stringify(packageTypeChoice)}`);
+
+        if (packageTypeChoice !== undefined && isPackageType(packageTypeChoice)){
+            setPackageType(req.session, packageTypeChoice);
+            return { url: PrefixedUrls.UPLOAD };
+        }
+
+        logger.error(`Package choice ${JSON.stringify(packageTypeChoice)} invalid. Rendering error.`);
```

Comment:
I'm not sure we want to log an error in the case of a validation error, that's still normal flow.